### PR TITLE
clang-format: fix schema validation and update definitions

### PR DESCRIPTION
**Pull request by loowps**
According to CLion some values aren't inline with the schema validation.

- fixes the expected types and values
- changes I'd like to make and we can discuss if needed:
  - change AllowShortFunctionsOnASingleLine from All to None <- this way functions are always identifiable by their consistent body
  - change SpaceBeforeParens from NonEmptyParentheses to ControlStatements <- this way parentheses of a function call are always tied to the function name and can be better differentiated from control statements
  - add SeparateDefinitionBlocks: "Always" to always add a blank line between function definitions etc.
  - add InsertBraces: true to wrap single line control statements that yet have no braces <- this way all control statements are equal and can be easily extended by a second line

### CODE REVIEW

**Comment by sleiner** [comment]
All of that looks sensible 👍🏻

> change SpaceBeforeParens from NonEmptyParentheses to ControlStatements <- this way parentheses of a function call are always tied to the function name and can be better differentiated from control statements

I, too, think that this is easier on the eyes. The reason this was set to `NonEmptyParantheses` is that the formatter configuration originally tried to mimic JUCE's style as close as possible. However, in the mean time I had set some other deviations from the JUCE style, so this fine too 👌

---

One request though: Could you please split the type changes into a separate commit and then create a new commit that includes:

  - the actual style changes
  - the changes to the source code introduced by these changes
  - the reasoning for the style changes that you laid out in the PR description as part of your commit message

**Review by sleiner — COMMENTED** [review]
All of that looks sensible 👍🏻

> change SpaceBeforeParens from NonEmptyParentheses to ControlStatements <- this way parentheses of a function call are always tied to the function name and can be better differentiated from control statements

I, too, think that this is easier on the eyes. The reason this was set to `NonEmptyParantheses` is that the formatter configuration originally tried to mimic JUCE's style as close as possible. However, in the mean time I had set some other deviations from the JUCE style, so this fine too 👌

---

One request though: Could you please split the type changes into a separate commit and then create a new commit that includes:

  - the actual style changes
  - the changes to the source code introduced by these changes
  - the reasoning for the style changes that you laid out in the PR description as part of your commit message

**Comment by sleiner** [comment]
Also, this PR made me aware of the fact that the clang-format CI job was broken, so I fixed it 👍🏻 Thanks! :-)

**Review by sleiner — APPROVED** [review]

**Review by sleiner — CHANGES_REQUESTED** [review]
I would also like to add `IncludeBlocks: Regroup`

**Comment by loowps** [comment]
I reverted and committed the changes separately.
But I'm a little bit lost here as I'm used to squash merging to main on PR complete.
Seems like the quality gate is evaluated for all commits and not the final diff..

File: pizjuce/CpuRam/CpuRamEditor.h [hunk 1]
```diff
@@ -1,27 +1,36 @@
...
 
+// changing order of includes breaks the build
+// clang-format off
 #include <juce_gui_basics/juce_gui_basics.h>
 #include <juce_gui_extra/juce_gui_extra.h>
 
 #include "CpuRam.h"
 #include "cputime.h"
 
+// clang-format on
```
Comment (sleiner) [inline pizjuce/CpuRam/CpuRamEditor.h]: I do not like this. Let's rather fix the underlying bug 🐛

File: .github/workflows/format.yml [hunk 2]
```diff
@@ -30,6 +30,6 @@ jobs:
                   fetch-depth: 0
                   ref: ${{ github.event.pull_request.head.sha }}
             - name: Install clang-format
-              run: sudo ./.github/scripts/install-clang-format.sh 14
+              run: sudo ./.github/scripts/install-clang-format.sh 15
```
Comment (sleiner) [inline .github/workflows/format.yml]: This is really inconvenient for me as clang-format 15 is not currently available on homebrew :/ However, this is not your fault and homebrew _should_ offer the current version. I will see what I can do...

**Review by sleiner — COMMENTED** [review]
> I'm a little bit lost here as I'm used to squash merging to main on PR complete.
Seems like the quality gate is evaluated for all commits and not the final diff..

Yes, I do not usually suqash-merge, since it tends to make the commit log to coarse. However, I still prefer a clean commit log, so I let this job check every commit indeed.

As for this merge request, I prefer this grouping:

- Changes to satisfy the schema validation in `.clang-format` (none of which should influence the actual formatting).
- Actual, semantic changes to `.clang-format`, including a reformatting of all the files.
- One commit for each "actual", semantic change. This is especially important because the `clang-format` docs advise for careful reviews when setting `InsertBraces` to `true`.
- The fix for the compile error in CpuRamEditor.

As you said that you were lost, here is some guidance on how to do that:

```bash
cd /path/to/pizmidi  # Ensure that you are in the repository root

git merge main  # Ensure that you are on the latest main
git reset main  # Reset your "commit pointer" to the latest main branch, keeping your working tree as is
                # If you commit now, you would create a single "squash commit" that you could merge.

git checkout pizjuce pizmidi  # Discard all the changes to the actual code

# ... stage and commit the schema changes to .clang-format

git stash push -- .clang-format  # Make a backup of the "actual" changes to .clang-format

# Now, for each of the changed lines in the stash:
your-editor .clang-format  # Modify the line in .clang-format
find . -name "*.h" -o -name "*.cpp" -o -name "*.cc" -o -name "*.c" -not -path "build/*" | xargs clang-format -i  # Reformat the code
git commit

# Lastly, do the patch for the CpuRam plugin
```

File: pizjuce/CpuRam/CpuRamEditor.h (hunk 1 above)
Comment (loowps) [reply pizjuce/CpuRam/CpuRamEditor.h]: Error message was a little bit cryptic to me.. but seems like an include was missing..

**Comment by loowps** [comment]
Thanks for the explanation. I'm more a visual guy and use sourcetree, but now it seems to be more clear to me (had to enable force push). Quite a nice way to rewrite the git history on a feature branch. Thx!

Sadly it seems like the current main doesn't comply with the formatting and therefore the first commit already breaks the check?

File: .github/workflows/format.yml (hunk 2 above)
Comment (loowps) [reply .github/workflows/format.yml]: Yes sry, insert braces seems to be available only from 15 onwards.

**Comment by sleiner** [comment]
> I'm more a visual guy and use sourcetree

Me to actually (although I'm using [Fork](https://fork.dev) instead of SourceTree), but these steps are much easier to explain with the command line 😅

> Sadly it seems like the current main doesn't comply with the formatting

The formatting on `main` is correct... at least as far as clang-format 14 is concerned. The formatting changes from aa2bcd6f0da0d38a8c17a26182595367d6a8f3fc are only preset if you use clang-format 15. The solution for that is separating the clang-format update into a separate commit. Since that requires some changes to CI scripts (`git-clang-format` has changed its return values) and I also noticed another problem when stacking multiple changes to `.clang-format` within a single pull request, I have separated this into a separate pull request: #25. Once that gets merged, please rebase your branch unto `main` :-)

File: .github/workflows/format.yml (hunk 2 above)
Comment (sleiner) [reply .github/workflows/format.yml]: A [patch for homebrew](https://github.com/Homebrew/homebrew-core/pull/114281) is on the way, and for the meantime, I have compiled a version of clang-format 15 locally :-)

**Comment by loowps** [comment]
> Me to actually (although I'm using [Fork](https://fork.dev) instead of SourceTree), but these steps are much easier to explain with the command line 😅

Ah, nice! IIRC I used Fork for some time with the idea to replace ST and liked it even more! But when I got used the differences, they unexpectedly went commercial. So I ditched it again.

Explanation was really helpful!

> The formatting on `main` is correct... at least as far as clang-format 14 is concerned. The formatting changes from [aa2bcd6](https://github.com/sleiner/pizmidi/commit/aa2bcd6f0da0d38a8c17a26182595367d6a8f3fc) are only preset if you use clang-format 15. The solution for that is separating the clang-format update into a separate commit. Since that requires some changes to CI scripts (`git-clang-format` has changed its return values) and I also noticed another problem when stacking multiple changes to `.clang-format` within a single pull request, I have separated this into a separate pull request: #25. Once that gets merged, please rebase your branch unto `main` :-)

Thanks, good work! Was a little bit naive on my end to simply increment the version without looking for breaking changes, sry!
I rebased the branch accordinly.

File: pizjuce/CpuRam/CpuRamEditor.h (hunk 1 above)
Comment (sleiner) [reply pizjuce/CpuRam/CpuRamEditor.h]: Perfect 👌

**Comment by sleiner** [comment]
So, one more nitpick about the commit log 🙈 Currently, the first line of the message is really long for many commits. This leads to weirdness in a multitude of tools displaying the log. See GitHub as an example:

image

Could you please rephrase those to a short summary on the first line (<= 50 characters) and paragraphs with around 72 characters per line afterwards (see [this writeup](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html) for reference)? 

#### Example:

```
clang-format: Set InsertBraces to true

This wraps single line control statements that have no braces yet, so
all control statements are equal and can be easily extended by a second
line.
```

**Comment by loowps** [comment]
> So, one more nitpick about the commit log 🙈 Currently, the first line of the message is really long for many commits. This leads to weirdness in a multitude of tools displaying the log.

Np, I fully understand your request. With cherry picking and ammending the commit message it can be adapted without much effort.. but tbh overall feels kind of outdated to be restricted to a certain amount of chars even if the credo is the shorter the better

File: pizmidi/midiTranspose/midiTranspose.cpp [hunk 3]
```diff
@@ -132,91 +136,111 @@ void MidiTranspose::setParameter (VstInt32 index, float value)
...
+            }
             else
-                sprintf (text, "%d", getTranspose());
+            {
+                sprintf(text, "%d", getTranspose());
+            }
             break;
         case kRange:
-            sprintf (text, "� %d", roundToInt (MAX_TRANSPOSE * param[index]));
+            sprintf(text, "� %d", roundToInt(MAX_TRANSPOSE * param[index]));
```
Comment (sleiner) [inline pizmidi/midiTranspose/midiTranspose.cpp]:
```suggestion
            sprintf(text, "± %d", roundToInt(MAX_TRANSPOSE * param[index]));
```

File: pizmidi/midiSimpleLFO/midiSimpleLFO.cpp [hunk 4]
```diff
@@ -268,185 +289,265 @@ float MidiSimpleLFO::getParameter (VstInt32 index)
...
+            }
             else
-                strcpy (label, "");
+            {
+                strcpy(label, "");
+            }
             break;
         case kPhase:
-            strcpy (label, "�");
+            strcpy(label, "�");
```
Comment (sleiner) [inline pizmidi/midiSimpleLFO/midiSimpleLFO.cpp]:
```suggestion
            strcpy(label, "°");
```

**Review by sleiner — CHANGES_REQUESTED** [review]
I have checked all commits but the last one (1aeafe4fba914ce144a4d276901d3c27c9da628b) and found two problems that seem related to character encoding (the repo uses UTF-8 everywhere - any idea what went wrong on your system there, @loowps?).

Workflow suggestion for fixing the commit in question (85b660a5d649dbdd0cfd100cab8ba868d73faff3): [fixup + interactive rebase](https://www.mikulskibartosz.name/git-fixup-explained/) (a bit faster and more comfortable than cherry-picking 😊)

After that fix, I will review the last commit (as opposed to the previous commits, I will have to look through these changes file by file...)

File: pizmidi/midiTranspose/midiTranspose.cpp (hunk 3 above)
Comment (loowps) [reply pizmidi/midiTranspose/midiTranspose.cpp]: Strange, according to the git history those problems weren't introduced on my system and were already present (?).
Nevertheless applied the two changes to the commit.

Comment (sleiner) [reply pizmidi/midiTranspose/midiTranspose.cpp]: They were not introduced by the commit where you now have appended the fix to ([`clang-format: set InsertBraces to true`](https://github.com/sleiner/pizmidi/commit/1aeafe4fba914ce144a4d276901d3c27c9da628b)) but in another commit that is part of this merge request ([`clang-format: change SpaceBeforeParens`](https://github.com/sleiner/pizmidi/commit/85b660a5d649dbdd0cfd100cab8ba868d73faff3)). I have [pinpointed](https://github.com/sleiner/pizmidi/commit/85b660a5d649dbdd0cfd100cab8ba868d73faff3#r88448239) one of the changed lines. Please append the fixes to that commit 😊

**Review by sleiner — COMMENTED** [review]
4fbce2342da49c5b86d43915355953cd4ac4bc75 introduces some weirdnesses around comments; I'll just fix these myself and then go ahead and merge this 👍🏻

**Review by sleiner — APPROVED** [review]

**Pull request merged**